persistence-api/src/main/java/io/stargate/db/query/builder/QueryBuilderImpl.java

```diff
@@ -1445,23 +1501,48 @@ protected BuiltSelect selectQuery() {
       wtColumn = table.column(writeTimeColumn);
       allSelected.add(wtColumn);
     }
+    Column countColumn = null;
+    if (countColumnName != null) {
+      countColumn = table.column(countColumnName);
+      allSelected.add(countColumn);
+    }
+    Column maxColumn = null;
+    if (maxColumnName != null) {
+      maxColumn = table.column(maxColumnName);
+      allSelected.add(maxColumn);
+    }
+    Column minColumn = null;
+    if (minColumnName != null) {
+      minColumn = table.column(minColumnName);
+      allSelected.add(minColumn);
+    }
+    Column sumColumn = null;
+    if (sumColumnName != null) {
+      sumColumn = table.column(sumColumnName);
+      allSelected.add(sumColumn);
+    }
+    Column avgColumn = null;
+    if (avgColumnName != null) {
+      avgColumn = table.column(avgColumnName);
+      allSelected.add(avgColumn);
+    }
+
     builder.append("SELECT");
-    if (selectedColumns.isEmpty() && writeTimeColumn == null) {
+    if (selectedColumns.isEmpty() && allFunctionCallColumnsAreNull()) {
       builder.append("*");
     } else {
       builder
           .start()
           .addAll(selectedColumns)
-          .addIfNotNull(
-              wtColumn,
-              c -> {
-                builder.append("WRITETIME(").append(c).append(")");
-                if (writeTimeColumnAlias != null) {
-                  builder.append("AS").append(cqlName(writeTimeColumnAlias));
-                }
-              })
+          .addIfNotNull(wtColumn, c -> addWriteTime(builder, c))
+          .addIfNotNull(countColumn, c -> addCount(builder, c))
+          .addIfNotNull(maxColumn, c -> addMax(builder, c))
+          .addIfNotNull(minColumn, c -> addMin(builder, c))
+          .addIfNotNull(sumColumn, c -> addSum(builder, c))
+          .addIfNotNull(avgColumn, c -> addAvg(builder, c))
```

Comment:
Could we handle functions in a generic way to avoid repetition?

Reply:
suggestions welcome :) I've refactored the same code to `addFunctionCallWithAliasIfPresent` method

Reply:
I do not have a concrete suggestion, but basically I was thinking about converting the `selection` field from a `List<String>` to a `List<Selector>`, `Selector` being a private class that encapsulates both simple columns and functions (perhaps in sub-classes).

Reply:
I don't think that this PR adds a lot of duplication - It follows the same pattern. If we are not fine with the duplication in this class, I think we should handle that separately, in a different PR that focuses exclusively on refactoring and this one should be scoped to adding a new feature. Is that ok?

Reply:
Indeed, current impl. is fine with the current grammar from my POV.

However, if we support a grammar like `(column|function)*` then I think having more generic code for functions would be worth the effort.

Reply:
After some more code investigation, I see that adding an alias to a `Column` is far from trivial. It requires changing the `Column` class and all `columns()` API (also potentially adding a new type to the API). Please note that the columns API accepts a list of columns as well. This is problematic in this context when we want to have aliases (especially alias using the `as()` method after the previous call). 
Because of those reasons, I think we should not implement aliases for columns in this ticket. At this point, we should not invest more time into it, especially because there is no demand for this feature.